src/schemes/structure/total_lagrangian_sph/system.jl

```diff
@@ -137,35 +137,40 @@ function TotalLagrangianSPHSystem(initial_condition, smoothing_kernel, smoothing
         @assert allunique(clamped_particles) "`clamped_particles` contains duplicate particle indices"
 
         n_clamped_particles = length(clamped_particles)
-        move_particles_to_end!(initial_condition, collect(clamped_particles))
-        move_particles_to_end!(young_modulus, collect(clamped_particles))
-        move_particles_to_end!(poisson_ratio, collect(clamped_particles))
+        ic_sorted_indices = deepcopy(initial_condition)
+        young_modulus_sorted_indices = copy(young_modulus)
+        poisson_ratio_sorted_indices = copy(poisson_ratio)
+        move_particles_to_end!(ic_sorted_indices, collect(clamped_particles))
```

Comment:
Why do you need the `collect`? This should work perfectly fine with just a range without allocating an extra vector.